src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -487,8 +487,16 @@ public override long Position
                 {
                     ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.value, ExceptionResource.ArgumentOutOfRange_NeedNonNegNum);
                 }
+                else if (_strategy.IsClosed)
+                {
+                    ThrowHelper.ThrowObjectDisposedException_FileClosed();
+                }
+                else if (!CanSeek)
+                {
+                    ThrowHelper.ThrowNotSupportedException_UnseekableStream();
+                }
 
-                _strategy.Seek(value, SeekOrigin.Begin);
+                _strategy.Position = value;
```

Comment:
`BufferedFileStreamStrategy` is used when buffering is enabled (`bufferSize > 1`), but when it's disabled then a class that derives from `OSFileStreamStrategy` is used.

`OSFileStreamStrategy.Seek` verifies that an existing content of a file opened for append is never overwritten by setting the file position:

https://github.com/dotnet/runtime/blob/372adbf8a4fcfe788e8f0fca3ed0fb9423099a76/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/OSFileStreamStrategy.cs#L170-L175

`OSFileStreamStrategy.Position` does not perform such check:

https://github.com/dotnet/runtime/blob/372adbf8a4fcfe788e8f0fca3ed0fb9423099a76/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/OSFileStreamStrategy.cs#L88

So we can't use `_strategy.Position` here.